### ISSUE TITLE
[css-values-5] Trailing comma in `attr()`

### ISSUE THREAD

**Issue by cdoublev** [w3c/csswg-drafts#8387]
Is a trailing comma allowed in [`attr()`](https://drafts.csswg.org/css-values-5/#attr-notation) like in [`var()`](https://drafts.csswg.org/css-variables-2/#using-variables)?

  > `attr() = attr( <q-name> <attr-type>? , <declaration-value>?)`

  > `var() = var( <custom-property-name> , <declaration-value>? )`
  > 
  > In an exception to the usual comma elision rules, which require commas to be omitted when they’re not separating values, a bare comma, with nothing following it, must be treated as valid in `var()`, indicating an empty fallback value.

Related: #6345

**Comment by tabatkins** [comment]
No, that's a special exception specifically for `var()` so you can substitute with an empty value (since you're allowed to *define* a custom property to have an empty value). It doesn't apply to anything else; the normal comma-elision rules are in force for attr().

**Comment by cdoublev** [comment]
But @tabatkins, custom properties can also be defined with `attr()`, `env()`, `paint()`, `random-item()`, `mix()`, which are all defined with `, <declaration-value>?` or `, <whole-value>`.

These [tests](https://github.com/web-platform-tests/wpt/blame/c2d015181a4956b2f91061595ec7da93823ec3ec/css/css-env/env-parsing.html#L12-L13) for `env()` allows a trailing comma.

What am I missing?

**Comment by tabatkins** [comment]
Well, none of those were specified to also have that special exception. It's fine if we do want to allow it, tho. ^_^

**Comment by cdoublev** [comment]
So I think it would be great to make an exception for `<declaration-value>?` or `<whole-value>` instead of in some specific functions.

**Comment by cdoublev** [comment]
f7a0776 allows a trailing `,` in `attr()` and `inherit()`. So this issue can perhaps be closed, unless this behavior should be generalized as suggested above.
